ext/pdo/pdo_stmt.c

```diff
@@ -2027,8 +2028,15 @@ static zend_function *dbstmt_method_get(zend_object **object_pp, zend_string *me
 static HashTable *dbstmt_get_gc(zend_object *object, zval **gc_data, int *gc_count)
 {
 	pdo_stmt_t *stmt = php_pdo_stmt_fetch_object(object);
-	*gc_data = &stmt->fetch.into;
-	*gc_count = 1;
+
+	if (stmt->default_fetch_type & PDO_FETCH_INTO) {
```

Comment:
A bitwise AND check seems wrong, `PDO_FETCH_INTO` is part of an enum with sequential integer indices so that means that there will be fetch mode with overlapping bits.
That said, I also think there's a pre-existing bug here: because `into` is part of a union, we should indeed only add the object/zval to the gc buffer if it is in this mode...

Reply:
Because of how the current struct is organized, this "into" zval also handles the class fetch ctor arguments.

Which means that I do have a bug indeed, thanks for making me realize this, I guess time to write even _more_ tests to handle these cases.

Reply:
Right, but the zval also overlap with an int, which is the bigger problem.